controllers/gemini.controller.js

```diff
@@ -33,29 +34,41 @@ const trackMood = async (req, res) => {
     }
 
     const mood = getMood(scale);
-    const requestData = {
-      contents: [
-        {
-          parts: [
+    const cacheKey = `mood_${scale}`;
```

Comment:
I didn't get your point here. The sole purpose of caching is to reduce the api calls, not to cache whatever you want. Please understand the concepts before implementing something. Here, the mood is dynamic and it will fetch different responses based on different moods selected. Why is there a need to cache? @tulu-g559